bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/DateTimeType.java

```diff
@@ -248,4 +250,85 @@ private ZonedDateTime parse(String value) throws DateTimeParseException {
 
         return date;
     }
+
+    public boolean isToday() {
+        return sameDay(ZonedDateTime.now());
+    }
+
+    public boolean isTomorrow() {
+        return sameDay(ZonedDateTime.now().plusDays(1));
+    }
+
+    public boolean isYesterday() {
+        return sameDay(ZonedDateTime.now().minusDays(1));
+    }
+
+    public boolean sameDay(DateTimeType other) {
+        return sameDay(other.zonedDateTime);
+    }
+
+    public boolean sameDay(ZonedDateTime other) {
+        return zonedDateTime.truncatedTo(ChronoUnit.DAYS)
+                .isEqual(other.withZoneSameInstant(zonedDateTime.getZone()).truncatedTo(ChronoUnit.DAYS));
```

Comment:
```suggestion
        return zonedDateTime.truncatedTo(ChronoUnit.DAYS).isEqual(other.truncatedTo(ChronoUnit.DAYS));
```

isEqual already compares the instant so it's not necessary to equalise the timezone prior to comparison.

```
jshell> var d1 = java.time.ZonedDateTime.parse("2024-01-01T00:00:00Z")
d1 ==> 2024-01-01T00:00Z

jshell> var d2 = java.time.ZonedDateTime.parse("2024-01-01T10:00:00+10:00")
d2 ==> 2024-01-01T10:00+10:00

jshell> d1.isEqual(d2)
$17 ==> true

jshell> d1.getZone()
$18 ==> Z

jshell> d2.getZone()
$19 ==> +10:00
```

Reply:
Changed